tests/parsers/test_cp.py

```diff
@@ -1,38 +1,62 @@
 # -*- coding: utf-8 -*-
 # pylint: disable=redefined-outer-name
 """Tests for the `CpParser`."""
+#import pdb
 import pytest
 
 from aiida import orm
 from aiida.common import AttributeDict
 
-
-@pytest.fixture
-def generate_inputs(generate_structure):
-    """Return only those inputs that the parser will expect to be there."""
-    return AttributeDict({
-        'structure': generate_structure(),
-        'parameters': orm.Dict(dict={}),
-    })
+import numpy
 
 
+#Now I use a post-6.5 git version (order of atom in the output changed)
+@pytest.mark.parametrize('version', ['default', '6.5_autopilot', '6.5', '6.5_cgstep', '6.5_cgsteps'])
 def test_cp_default(
-    aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, generate_inputs, data_regression
+    aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression, generate_structure,
+    version, call_something
 ):
     """Test a default `cp.x` calculation."""
     entry_point_calc_job = 'quantumespresso.cp'
     entry_point_parser = 'quantumespresso.cp'
+    if version == 'default':
+
+        def generate_inputs():
+            return AttributeDict({
+                'structure': generate_structure(structure_id='silicon'),
+                'parameters': orm.Dict(dict={}),
+            })
+    else:
+
+        def generate_inputs():
+            return AttributeDict({
+                'structure': generate_structure(structure_id='water'),
+                'parameters': orm.Dict(dict={}),
+            })
 
-    node = generate_calc_job_node(entry_point_calc_job, fixture_localhost, 'default', generate_inputs)
+    node = generate_calc_job_node(entry_point_calc_job, fixture_localhost, version, generate_inputs())
     parser = generate_parser(entry_point_parser)
     results, calcfunction = parser.parse_from_node(node, store_provenance=False)
 
     assert calcfunction.is_finished, calcfunction.exception
     assert calcfunction.is_finished_ok, calcfunction.exit_message
-    assert not orm.Log.objects.get_logs_for(node)
+    if version != '6.5_cgstep':
```

Comment:
From this line below, we suggest to simplify the logic as follows:
```python
assert 'output_parameters' in results 
data = {'parameters': results['output_parameters'],}

if version != '6.5_cgstep': 
    assert not orm.Log.objects.get_logs_for(node)
    assert 'output_trajectory' in results 
    data['output_trajectory'] = results['output_trajectory']

data_regression(data) 
```

Reply:
modified in commit f2f78da3